test/integration/template_test.rb

```diff
@@ -323,4 +323,30 @@ def test_using_range_literal_works_as_expected
     result = t.render('x' => 1, 'y' => 5)
     assert_equal('12345', result)
   end
+
+  def test_instrument_for_bug_1346
+    calls = []
+    Liquid::Usage.stub(:increment, ->(name) { calls << name }) do
+      Liquid::Template.parse("{% for i in (1..2) %}{{ i }}{% endfor {% foo %}")
+    end
+    assert_equal(["multiple_open_tags"], calls)
+
+    calls = []
+    Liquid::Usage.stub(:increment, ->(name) { calls << name }) do
+      Liquid::Template.parse("{% for i in (1..2) %}{{ i }}{% endfor {% 123 %}")
+    end
+    assert_equal(["multiple_open_tags"], calls)
+
+    calls = []
+    Liquid::Usage.stub(:increment, ->(name) { calls << name }) do
+      Liquid::Template.parse("{% for i in (1..2) %}{{ i }}{% endfor %}")
+    end
+    assert_equal([], calls)
+
+    calls = []
+    Liquid::Usage.stub(:increment, ->(name) { calls << name }) do
+      Liquid::Template.parse("{% for i in (1..2) %}{{ i }}{% endfor test %}")
+    end
+    assert_equal([], calls)
```

Comment:
This is an example of the sort of thing we want to catch with the instrumentation and is the type of thing that it looks like we accidentally started allowing for the raw tag in liquid-c.